src/main/scala/temple/generate/server/go/service/GoServiceGenerator.scala

```diff
@@ -17,22 +18,31 @@ object GoServiceGenerator extends ServiceGenerator {
      * handlers in <>.go
      * config.json
      */
-    val usesComms  = root.comms.nonEmpty
+
+    // Set of CRUD operations the service uses
     val operations = root.opQueries.keySet
+
+    // Whether or not the service uses inter-service communication
+    val usesComms = root.comms.nonEmpty
+
+    // Whether or not ths service uses the time type, by checking for attributes of type date, time or datetime
+    val usesTime =
+      Set[AttributeType](AttributeType.DateType, AttributeType.TimeType, AttributeType.DateTimeType)
+        .intersect(root.attributes.values.map(_.attributeType).toSet)
+        .nonEmpty
+
     (Map(
       File(s"${root.name}", "go.mod") -> GoCommonGenerator.generateMod(root.module),
       File(root.name, s"${root.name}.go") -> mkCode.doubleLines(
         GoCommonGenerator.generatePackage("main"),
-        GoServiceMainGenerator.generateImports(root, usesComms),
-        GoServiceMainGenerator.generateGlobals(root, usesComms),
+        GoServiceMainGenerator.generateImports(root, usesTime, usesComms),
+        GoServiceMainGenerator.generateEnvStruct(usesComms),
         GoServiceMainGenerator.generateMain(root, usesComms, operations),
-        GoCommonMainGenerator.generateJsonMiddleware(),
-        GoServiceMainGenerator.generateHandlers(root, operations),
```

Comment:
Do we still want this function? It’s now untested

Reply:
Will be used in the PR after next